### PR TITLE
MWPW-153245 [merch][analytics] dispatch wcomp events, and let default lh

### PR DESCRIPTION
**Pull request by npeltier**
- dispatch merch wc shadow event for analytics,
- if lh attribute is already set, keep it, appending eventually mep information,
- also remove merch-sidenav dependency that is only consumed in CC atm,

Resolves: [MWPW-136881](https://jira.corp.adobe.com/browse/MWPW-136881)

mas PR: https://github.com/adobecom/mas/pull/42

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.live/?martech=off
- After: https://mwpw-153245--milo--npeltier.hlx.live/?martech=off

- Before: https://main--cc--adobecom.hlx.live/products/catalog
- After: https://mwpw-153245--cc--npeltier.hlx.live/products/catalog?milolibs=mwpw-153245--milo--npeltier

### CODE REVIEW

**Comment by npeltier** [comment]
no "need" to resolve conflicts right now, as dep will be overwritten one last time once first approvals are made

**Review by Axelcureno — APPROVED** [review]

File: libs/martech/attributes.js [hunk 1]
```diff
@@ -73,9 +73,14 @@ export async function decorateSectionAnalytics(section, idx, config) {
   });
   const mepMartech = config?.mep?.martech || '';
   section.querySelectorAll('[data-block]').forEach((block, blockIdx) => {
-    const blockName = block.classList[0] || '';
-    block.setAttribute('daa-lh', `b${blockIdx + 1}|${blockName.slice(0, 15)}${mepMartech}`);
-    decorateDefaultLinkAnalytics(block, config);
+    const lhAtt = block.getAttribute('daa-lh');
+    if (!lhAtt) {
```
Comment (chrischrischris) [inline libs/martech/attributes.js:77]: nit: reverse if/else: `if (lhAtt) {`

**Review by chrischrischris — APPROVED** [review]

File: libs/martech/attributes.js [hunk 2]
```diff
@@ -73,9 +73,14 @@ export async function decorateSectionAnalytics(section, idx, config) {
...
+      const blockName = block.classList[0] || '';
+      block.setAttribute('daa-lh', `b${blockIdx + 1}|${blockName.slice(0, 15)}${mepMartech}`);
+      decorateDefaultLinkAnalytics(block, config);
+    } else {
+      block.setAttribute('daa-lh', `${lhAtt}${mepMartech}`);
+    }
```
Comment (vgoodric) [inline libs/martech/attributes.js:83]:
```suggestion
    let lhAtt = block.getAttribute('daa-lh');
    if (!lhAtt) {
      const blockName = block.classList[0] || '';
      lhAtt = `b${blockIdx + 1}|${blockName.slice(0, 15)}`;
    }
    block.setAttribute('daa-lh', `${lhAtt}${mepMartech}`);
    decorateDefaultLinkAnalytics(block, config);
```
Need to keep decorateDefaultLinkAnalytics outside of the if statement.  And I think this suggestion is cleaner.

**Review by vgoodric — CHANGES_REQUESTED** [review]

**Comment by Roycethan** [comment]
@npeltier Can you plz review below:
1) Sort:Popularity event name is "authored--:sort|s2|all|nopzn|catalog-devicet" doesn't have keyword Popularity
2) Hover over action menu click on any card generates same event name without any product name i.e. 
"merch-card:action-menu-toggle|s2|all|nopzn|catalog-devicet"

**Comment by npeltier** [comment]
@Roycethan can you have a second look? (not sure why but search related network call happens much after it, but still does)

**Comment by Roycethan** [comment]
> @npeltier Can you plz review below:
> 
> 1. Sort:Popularity event name is "authored--:sort|s2|all|nopzn|catalog-devicet" doesn't have keyword Popularity
> 2. Hover over action menu click on any card generates same event name without any product name - for all the product cards i.e.
>    "merch-card:action-menu-toggle|s2|all|nopzn|catalog-devicet"
> 3. Search box - search doesn't triggered any events
> 4. Show more button not triggering analytics events - can be checked at https://main--cc--adobecom.hlx.live/products/catalog?milolibs=mwpw-153245--milo--npeltier#category=all
> 5. Product icon in the card which is clickable , same case with product name -
>    6)Learn more (link) - has daa-ll , but event name is not populated
> 
> Apart from above observation/issues , covered regressions.

@npeltier 1,2,4 are fixed... 3 ,5,6 are open

**Review by vgoodric — APPROVED** [review]
Checked the update.  Looks good to me

**Comment by Roycethan** [comment]
> > @npeltier Can you plz review below:
> > 
> > 1. Sort:Popularity event name is "authored--:sort|s2|all|nopzn|catalog-devicet" doesn't have keyword Popularity
> > 2. Hover over action menu click on any card generates same event name without any product name - for all the product cards i.e.
> >    "merch-card:action-menu-toggle|s2|all|nopzn|catalog-devicet"
> > 3. Search box - search doesn't triggered any events
> > 4. Show more button not triggering analytics events - can be checked at https://main--cc--adobecom.hlx.live/products/catalog?milolibs=mwpw-153245--milo--npeltier#category=all
> > 5. Product icon in the card which is clickable , same case with product name -
> >    6)Learn more (link) - has daa-ll , but event name is not populated
> > 
> > Apart from above observation/issues , covered regressions.
> 
> @npeltier 1,2,4,5,6 are fixed... 3 is fixed

**Pull request merged**